iree/compiler/Dialect/Flow/Transforms/Passes.cpp

```diff
@@ -145,6 +145,9 @@ void buildFlowTransformPassPipeline(OpPassManager &passManager,
   // executables.
   passManager.addPass(createOutlineDispatchRegionsPass());
 
+  // Generate tag for each executable and store it in an attribute.
+  passManager.addNestedPass<mlir::ModuleOp>(createGenerateDispatchTagsPass());
```

Comment:
My other comment mentioned the full range of where this pass _could_ go, but I think I'd specifically suggest before the "stream formation" block below. @MaheshRavishankar or @benvanik may have other suggestions for point where the executables are defined well enough that summarizing them at that point would be most useful for future analysis (e.g. in Tracy).

The cleanup and deduplication passes between this point and the "stream formation" cut down on the IR complexity quite a bit. Here's a visualization from ~5 months ago: https://scotttodd.github.io/iree-llvm-sandbox/web-tools/pipeline-visualizer/?data=./data/bert_encoder_llvmaot.json. On the left side of the chart you can see that there are 37000 ops in the top level module before `DeduplicateExecutables` and 2700 ops after.